File: experiments/source/python/affine_score_reduce_reordered.py
import loom


@loom.entry
def affine_score_reduce_reordered(scale: float, bias: float, threshold: float, x: loom.Tensor1) -> float:
    def score(xi: float) -> float:
        shifted = bias + (xi * scale)
        return shifted if threshold < shifted else 0.0

    return loom.Tensor1.reduce_sum(loom.Tensor1.map(score, x))

File: experiments/source/ocaml/affine_score_reduce_reordered.ml
open Loom

let[@loom.entry] affine_score_reduce_reordered (scale : float) (bias : float)
    (threshold : float) (x : Tensor1.t) =
  let score xi =
    let shifted = bias +. (xi *. scale) in
    if threshold < shifted then shifted else 0.0
  in
  Tensor1.reduce_sum (Tensor1.map score x)
